membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -168,12 +173,16 @@ case class RatePlan(
     productName: String,
     lastChangeType: Option[String],
     features: List[Feature],
-    chargedThroughDate: Option[LocalDate],
     ratePlanCharges: NonEmptyList[RatePlanCharge],
-    start: LocalDate,
-    end: LocalDate,
 ) {
 
+  def chargedThroughDate: Option[LocalDate] =
+    ratePlanCharges.map(_.chargedThroughDate).list.toList.max // None is less than Some
+  def effectiveStartDate: LocalDate =
+    ratePlanCharges.map(_.effectiveStartDate).list.toList.min
+  def effectiveEndDate: LocalDate =
+    ratePlanCharges.map(_.effectiveEndDate).list.toList.min
```

Comment:
these helper methods replicate the old logic that used to be in the deserialiser, (not to mention it was pretty sensible, it just didn't match the zuora data)